docs/source/analyze/mysql.md

```diff
@@ -0,0 +1,64 @@
+---
+title: MySQL
+description: Check version and connection status 
+---
+
+The `MySQL` analyzer is available to check version and connection status of a MySQL database. It relies on the data collected by the [MySQL collector](https://troubleshoot.sh/docs/collect/mysql/).
+
+The analyzer's outcome `when` clause may be used to evaluate the database connection status or a semver range to compare against the running version, and supports standard comparison operators.
+
+## Paramaters
+
+**checkName:** Optional name.
+
+**collectorName:** (Required) Must be "mysql".
+
+## Outcomes
+
+The `when` value in an outcome of this analyzer contains the connection or version information.
+
+The conditional in the when value supports the following:
+
+**connected:** A boolean representing whether the database is connected. Can be compared to a boolean value with the `==` operator.
+
+**version:** A string representing the semantic version of the database. Can be compared to a semver string using `<`, `<=`, `>`, `>=`, `==`, `!=`, with the letter 'x' as a version wildcard (10.x). The 'x' is parsed as '0'.
+  
+## Example Analyzer Definition
+
+```yaml
+apiVersion: troubleshoot.sh/v1beta2
+kind: Preflight
+metadata:
+  name: supported-mysql-version
+spec:
+  collectors:
+    - mysql:
+        collectorName: mysql
+        uri: '<user>:<password>@tcp(<host>:<port>)/<dbname>'
+  analyzers:
+    - mysql:
+        checkName: Must be MySQL 8.x or later
+        collectorName: mysql
```

Comment:
@manavellamnimble Does the analyzer really need the collectorName specified?

Reply:
@markpundsack only if the collector also has a `collectorName`. 
The **collector** will save the output in mysql/`collectorName`.json, and collectorName defaults to "mysql". The **analyzer** will look for this file in mysql/`collectorName`.json, and it also defaults to "mysql".

That means, if both the analyzer and the collector have no name, it is ok (though it may lead to data loss if more than one mysql collector doesn't have a collectorName. If the collector has a collectorName, the analyzer must have the same collectorName.

That being said, I would change the mysql collector [docs](https://troubleshoot.sh/docs/collect/mysql/) to say that collectorName is a required and not just recommended field.

Reply:
Thanks. FWIW, it's not enough to change docs to say it's required if the code doesn't actually make it required. That would be a breaking change, so let's not do that now.

Reply:
No sure, that would imply modifying the code, I could take care of that. Just let me know!